accel/tcg/tcg-all.c

```diff
@@ -54,8 +54,8 @@ static void tcg_handle_interrupt(CPUState *cpu, int mask)
     int old_mask;
     g_assert(qemu_mutex_iothread_locked());
 
-    old_mask = cpu->interrupt_request;
-    cpu->interrupt_request |= mask;
+    old_mask = atomic_read(&cpu->interrupt_request);
+    atomic_set(&cpu->interrupt_request, cpu->interrupt_request | mask);
```

Comment:
You reading old_mask, bur ORing with mask.
Would it be better to use atomic OR ?